app/components/LogoutWarningModal.tsx

```diff
@@ -0,0 +1,53 @@
+import React, {useEffect, useState} from 'react';
+import {Button, Container, Form, Modal, Row} from 'react-bootstrap';
+
+interface Props {
+  inactivityDelaySeconds: number;
+  expiresOn: number;
+  onExtendSession: () => void;
+}
+
+const LogoutWarningModal: React.FunctionComponent<Props> = ({
+  inactivityDelaySeconds,
+  expiresOn,
+  onExtendSession
+}) => {
+  const [remainingSeconds, setRemainingSeconds] = useState(
+    Math.floor((expiresOn - Date.now()) / 1000)
+  );
+
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      setRemainingSeconds(Math.floor((expiresOn - Date.now()) / 1000));
+    }, 1000);
+
+    return () => {
+      clearInterval(intervalId);
+    };
+  }, []);
+
+  return (
+    <Modal show size="lg" onHide={() => {}}>
```

Comment:
```suggestion
    <Modal show size="lg">
```
Is the noop `onHide` needed?

Reply:
no!